sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/exporter/SpanExporter.scala

```diff
@@ -0,0 +1,120 @@
+import cats.syntax.parallel._
+import org.typelevel.otel4s.sdk.trace.data.SpanData
+
+/** An interface that allows different tracing services to export recorded data
+  * for sampled spans in their own format.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/trace/sdk/#span-exporter]]
+  *
+  * @tparam F
+  *   the higher-kinded type of a polymorphic effect
+  */
+trait SpanExporter[F[_]] {
+
+  /** Called to export sampled [[data.SpanData SpanData]].
+    *
+    * '''Note''': the export operations can be performed simultaneously
+    * depending on the type of span processor being used. However, the batch
+    * span processor will ensure that only one export can occur at a time.
+    *
+    * @param spans
+    *   the collection of sampled spans to be exported
+    */
+  def exportSpans(spans: List[SpanData]): F[Unit]
+
+  /** Exports the collection of sampled [[data.SpanData SpanData]] that have not
+    * yet been exported.
+    *
+    * '''Note''': the export operations can be performed simultaneously
+    * depending on the type of span processor being used. However, the batch
+    * span processor will ensure that only one export can occur at a time.
+    */
+  def flush: F[Unit]
+}
+
+object SpanExporter {
+
+  /** Creates a [[SpanExporter]] which delegates all exports to the exporters.
+    */
+  def of[F[_]: Applicative: Parallel](
+      exporters: SpanExporter[F]*
+  ): SpanExporter[F] =
+    if (exporters.sizeIs == 1) exporters.head
+    else exporters.combineAll
+
+  /** Creates a no-op implementation of the [[SpanExporter]].
+    *
+    * All export operations are no-op.
+    */
+  def noop[F[_]: Applicative]: SpanExporter[F] =
+    new Noop
+
+  implicit def spanExporterMonoid[F[_]: Applicative: Parallel]
+      : Monoid[SpanExporter[F]] =
+    new Monoid[SpanExporter[F]] {
+      val empty: SpanExporter[F] =
+        noop[F]
+
+      def combine(x: SpanExporter[F], y: SpanExporter[F]): SpanExporter[F] =
+        (x, y) match {
+          case (that, _: Noop[F]) =>
+            that
+          case (_: Noop[F], other) =>
+            other
+          case (that: Multi[F], other: Multi[F]) =>
+            Multi(that.exporters.concatNel(other.exporters))
+          case (that: Multi[F], other) =>
+            Multi(that.exporters :+ other)
+          case (that, other: Multi[F]) =>
+            Multi(that :: other.exporters)
+          case (that, other) =>
+            Multi(NonEmptyList.of(that, other))
+        }
+    }
+
+  private final class Noop[F[_]: Applicative] extends SpanExporter[F] {
+    def exportSpans(spans: List[SpanData]): F[Unit] = Applicative[F].unit
+    def flush: F[Unit] = Applicative[F].unit
+
+    override def toString: String = "SpanExporter.Noop"
+  }
+
+  private final case class Multi[F[_]: Parallel](
+      exporters: NonEmptyList[SpanExporter[F]]
+  ) extends SpanExporter[F] {
+    def exportSpans(spans: List[SpanData]): F[Unit] =
+      exporters.parTraverse_(_.exportSpans(spans))
```

Comment:
Currently this fails fast. If any of the exporters fails, then the rest will be canceled. I'm not sure if that's the behavior we want.

We might want to use `attempt` and gather the failures into `CompositeFailure`, if there is more than one?

Reply:
That makes sense.

How should we encode the `CompositeFailure`? Should we take `exporter name` into the account?

I've describe possible solutions in the description, see `2) export must return the ExportResult` (GitHub doesn't allow ref link to the description 😞  )

Reply:
I agree it's not worth the hustle /shrug the spec doesn't say anything about this right?

Reply:
Let's say we have 10 exporters. 2 complete with failures and 8 work fine.
I don't think we should cut the other 8 healthy exporters off.
As you suggested, we can use `attempt` while evaluating each `exportSpans(spans)`.  And in the end, we can throw a compose failure (if any). The healthy exports will do their job, while the end user will still have enough details about the failures.
Well, it is kinda far away from being ideal or reasonable, but it still may work.

We can also consider using `Ior` type.

Reply:
Oh, in case it wasn't clear, `CompositeFailure` is an FS2 thing.

https://www.javadoc.io/doc/co.fs2/fs2-docs_2.13/latest/fs2/CompositeFailure.html

Reply:
Ah, I completely forgot. Thanks!

Reply:
Hm, we need to bring `fs2-core` dependency then. I'm not sure if it's worth it for a single exception.

Reply:
Aha, okay, I couldn't remember if FS2 was already a dependency of this module.

Eh. In that case I say we just raise the first exception or something, idk.

Reply:
If we want to keep exporters as a separate module, fs2 dependency is not needed. If we want to have exporters as a part of the sdk module, we can incorporate fs2-core

Reply:
I added a variation of the `CompositeFailure`.